receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go

```diff
@@ -0,0 +1,334 @@
+				require.NoError(t, err)
+				filterCasted := filter.(*itemCardinalityFilter)
+				defer executeShutdown(t, filterCasted)
+
+				assert.Equal(t, metricName, filterCasted.metricName)
+				assert.Equal(t, testCase.totalLimit, filterCasted.totalLimit)
+				assert.Equal(t, testCase.limitByTimestamp, filterCasted.limitByTimestamp)
+				assert.Equal(t, itemActivityPeriod, filterCasted.itemActivityPeriod)
+				assert.Equal(t, logger, filterCasted.logger)
+				require.NotNil(t, filterCasted.cache)
+			}
+		})
+	}
+}
+
+func TestItemCardinalityFilter_TotalLimit(t *testing.T) {
+	itemFilter := &itemCardinalityFilter{totalLimit: totalLimit}
+
+	assert.Equal(t, totalLimit, itemFilter.TotalLimit())
+}
+
+func TestItemCardinalityFilter_LimitByTimestamp(t *testing.T) {
+	itemFilter := &itemCardinalityFilter{limitByTimestamp: limitByTimestamp}
+
+	assert.Equal(t, limitByTimestamp, itemFilter.LimitByTimestamp())
+}
+
+func TestItemCardinalityFilter_CanIncludeNewItem(t *testing.T) {
+	logger := zap.NewNop()
+	testCases := map[string]struct {
+		totalLimit         int
+		limitByTimestamp   int
+		keysAlreadyInCache []string
+		expectedResult     bool
+	}{
+		"No items in cache and timestamp limit hasn't been exhausted": {2, 1, nil, true},
+		"Cache is full and timestamp limit hasn't been exhausted":     {2, 1, []string{"qwerty1", "qwerty2"}, false},
+		"No items in cache and timestamp limit has been exhausted":    {2, 0, nil, false},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			filter, err := NewItemCardinalityFilter(metricName, testCase.totalLimit, testCase.limitByTimestamp,
+				itemActivityPeriod, logger)
+			require.NoError(t, err)
+			filterCasted := filter.(*itemCardinalityFilter)
+			defer executeShutdown(t, filterCasted)
+
+			for _, key := range testCase.keysAlreadyInCache {
+				err = filterCasted.cache.Set(key, byte(1))
+				require.NoError(t, err)
+			}
+
+			assert.Equal(t, testCase.expectedResult, filterCasted.canIncludeNewItem(testCase.limitByTimestamp))
+		})
+	}
+}
+
+func TestItemCardinalityFilter_Shutdown(t *testing.T) {
+	logger := zap.NewNop()
+	testCases := map[string]struct {
+		closeCache  bool
+		expectError bool
+	}{
+		"Happy path":            {false, false},
+		"Cache has been closed": {true, true},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			filter, err := NewItemCardinalityFilter(metricName, totalLimit, limitByTimestamp, itemActivityPeriod, logger)
+			require.NoError(t, err)
+			filterCasted := filter.(*itemCardinalityFilter)
+
+			if testCase.closeCache {
+				// Covering case when by some reasons cache is closed
+				err = filterCasted.cache.Close()
+				require.NoError(t, err)
+			}
+
+			if testCase.expectError {
+				require.Error(t, filter.Shutdown())
+			} else {
+				require.NoError(t, filter.Shutdown())
+			}
+		})
+	}
+}
+
+func TestItemCardinalityFilter_Filter(t *testing.T) {
+	items := initialItems(t)
+	logger := zap.NewNop()
```

Comment:
I suggest to use `zaptest.NewTestLogger(t)`

Reply:
Done.